api-server.py

```diff
@@ -101,9 +118,37 @@ def main():
     elif args.coznet:
         settings.setup_coznet()
 
+    syslog_facility = None
+    if args.syslog or args.syslog_local:
+        # Setup the syslog facility
+        if args.syslog_local:
+            print("Logging to syslog local facility %s" % args.syslog_local)
+            syslog_facility = SysLogHandler.LOG_LOCAL0 + args.syslog_local
+        else:
+            print("Logging to syslog user facility")
+            syslog_facility = SysLogHandler.LOG_USER
+
+        # Setup logzero to only use the syslog handler
+        logzero.logfile(None, disableStderrLogger=args.disable_stderr)
+        syslog_handler = SysLogHandler(facility=syslog_facility)
+        logger.addHandler(syslog_handler)
+    else:
+        # Setup file logging
+        if args.logfile:
+            logfile = os.path.abspath(args.logfile)
+            print("Logging to logfile and stdout: %s" % logfile)
+            logzero.logfile(logfile, maxBytes=LOGFILE_MAX_BYTES, backupCount=LOGFILE_BACKUP_COUNT, disableStderrLogger=args.disable_stderr)
+
+        else:
+            print("Logging to stdout and stderr")
+
     # Write a PID file to easily quit the service
     write_pid_file()
 
+    # Setup twisted / klein to reuse the logzero setup
+    observer = log.PythonLoggingObserver(loggerName=logzero.LOGZERO_DEFAULT_LOGGER)
+    observer.start()
```

Comment:
I'm not fully understanding how this is working (or supposed to work). In my testing, the Klein logging still isn't going to the logfile (when logfile is enabled). I see the normal blockchain logging in the logfile, but nothing from Klein.

Reply:
Added a commit that should fix it. Would be great if you could test again.